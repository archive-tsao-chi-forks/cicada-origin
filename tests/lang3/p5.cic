Dup : Type
Dup = {
  T: Type
  x: T
  y: T
}

@show Dup

_1 : Dup
_1 = {
  T = String
  x = "abc"
  y = "123"
}

@show _1

_2 : Dup[String]
_2 = {
  T = String
  x = "abc"
  y = "123"
}

@show _2

_3 : Dup[String]["abc"]
_3 = {
  T = String
  x = "abc"
  y = "123"
}

@show _3

_4 : Dup[String]["abc"]["123"]
_4 = {
  T = String
  x = "abc"
  y = "123"
}

@show _4
