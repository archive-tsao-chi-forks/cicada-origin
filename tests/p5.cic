@def xyh "Xie Yuheng"

@show xyh
