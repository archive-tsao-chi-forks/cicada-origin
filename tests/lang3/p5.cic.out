Type -- {
  T : Type
  x : T
  y : T
}
{
  T : Type
  x : T
  y : T
} -- {
  T = String
  x = "abc"
  y = "123"
}
{
  T : Type = Stringx : String
  y : String
} -- {
  T = String
  x = "abc"
  y = "123"
}
